common/beerocks/hostapd/include/configuration.h

```diff
@@ -192,11 +203,25 @@ class Configuration {
 };
 
 template <class func> void Configuration::for_all_ap_vaps(func f)
+{
+    auto f_with_iter = [&f](const std::string &vap, int iter) { f(vap); };
+
+    int dummy(0);
+    for_all_ap_vaps(f_with_iter, dummy, 10);
+}
+
+template <class func, class iter>
+void Configuration::for_all_ap_vaps(func f, iter current_iter, const iter end)
```

Comment:
I would have turned it around, something like this:
```suggestion
template <class func, class iter>
void Configuration::for_all_ap_vaps(func f, iter current_iter, const iter end)
{
    for_all_ap_vaps([](const std::vector<std::string> &vap) {
        if (current_iter != end) {
            f(vap, current_iter);
            current_iter++;
        } else {
            /* Call the second function here for the remaining vaps */
        }
        });
```